src/iop/scalepixels.c

```diff
@@ -177,11 +196,10 @@ void process(dt_iop_module_t *self, const dt_dev_pixelpipe_iop_t *const piece, c
     float *out = ((float *)ovoid) + (size_t)4 * j * roi_out->width;
     for(int i = 0; i < roi_out->width; i++, out += 4)
     {
-      float po[2] = { i, j };
+      float x = MIN(i*d->x_scale, roi_in->width);
```

Comment:
`MIN()` is not needed.
Because, i believe, dt_interpolation_compute_pixel4c() won't sample out of buffer.
And i think because that would hide the issue with wrong `d->*_scale`